tokio/src/runtime/time/mod.rs

```diff
@@ -334,10 +352,16 @@ impl Handle {
                 if !waker_list.can_push() {
                     // Wake a batch of wakers. To avoid deadlock, we must do this with the lock temporarily dropped.
                     drop(lock);
+                    drop(wheels_lock);
```

Comment:
You also need to drop `wheels_lock` after the loop, since there's also a call to `wake_all` there.

Reply:
Sure! Missed that...